studio/src/editor-panel.js

```diff
@@ -186,14 +187,51 @@ export default class EditorPanel extends LitElement {
         );
     }
 
+    getFragmentPropsToUse() {
+        const props = {};
+        this.fragment?.fields.forEach((field) => {
+            if (field.name === 'cardTitle' && field.values?.length) {
+                props.cardTitle = field.values[0];
+            }
+            if (field.name === 'variant' && field.values?.length) {
+                props.variantCode = field.values[0];
+            }
+        });
+        VARIANTS.forEach((variant) => {
+            if (variant.value === props.variantCode) {
+                props.variantLabel = variant.label;
+                props.surface = variant.surface;
+            }
+        });
+        return props;
+    }
+
     async copyToUse() {
-        //@TODO make it generic.
-        const code = `<merch-card><aem-fragment fragment="${this.fragment?.id}"></aem-fragment></merch-card>`;
+        //@TODO make it generic also for merch-card-collection
+        const props = this.getFragmentPropsToUse();
+        const code = `<merch-card><aem-fragment fragment="${this.fragment?.id}" title="${props.cardTitle}"></aem-fragment></merch-card>`;
```

Comment:
I think we need a mapping between model name and web component name:
see: `fragment.model.name`, to keep it more genenric:
```js
{
"Card": "merch-card",
"Card Collection": "merch-card-collection" // not sure space in the name is valid
}
```

Reply:
Done. I added `'Card Collection': 'merch-card-collection'` as you proposed
if we have a different model name, will change it